src/constants/customer-form-config.js

```diff
@@ -33,20 +35,49 @@ export const fields = [
     name: 'id',
     label: 'id',
   },
+  {
+    name: 'logo',
+    label: 'logo',
+  },
+  {
+    name: 'file',
+    label: 'file',
+  },
 ];
 
 export const hooks = {
   onSuccess(form: Object) {
+    utilityStore.turnOnWaiting();
     // get initial values of the form, to see if we are creating or updating modalStyle
     const initialsValue = form.initials();
     if (
       initialsValue.name === '' &&
       initialsValue.url === '' &&
       initialsValue.industry === ''
     ) {
-      customersStore.createCustomer(form.values());
+      if (form.values().file) {
+        uploadImage(form.values().file)
+          .then(logoId => getImage(logoId))
+          .then(logoUrl => {
+            customersStore.createCustomer(
+              Object.assign({}, form.values(), {logo: logoUrl})
+            );
+          });
+      } else {
+        customersStore.createCustomer(form.values());
+      }
     } else {
-      customersStore.updateCustomer(form.values());
+      if (form.values().file !== '') {
```

Comment:
`if (form.values().file) {` is enough

Reply:
form from mobx-react-form automatically fills empty value with an empty string, so it is needed.